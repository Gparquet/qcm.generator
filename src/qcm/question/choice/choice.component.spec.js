import React from 'react';
import renderer from 'react-test-renderer';
import ChoiceComponent from './choice.component';
import choicesFormatted from './mocks/choices_of_question_formatted';
import choiceConstants from './choice.constants';

describe('Choice Component', ()=>{
  const props = {
    ...choiceConstants.fieldInformation,
    choices: choicesFormatted,
  };

  test('renders correctly', ()=>{
    const tree = renderer
      .create(<ChoiceComponent {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
