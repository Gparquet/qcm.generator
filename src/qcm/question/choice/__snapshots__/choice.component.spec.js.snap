// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choice Component renders correctly 1`] = `
<div
  className="row af-form__group"
>
  <div
    className="col-md-2"
  >
    <label
      className="af-form__group-label"
      htmlFor=""
    >
      Choices
    </label>
  </div>
  <div
    className="col-md-10"
  >
    <div
      className="af-form__radio-custom"
    >
      <input
        autoFocus={null}
        checked={false}
        className="af-form__input-radio"
        disabled={false}
        id="cjvhuyr8v0000jctd801b4zke"
        name="choice"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        tabIndex={null}
        type="radio"
        value="5f67c896-77b4-4b9b-bf19-fb2c8e274c68"
      />
      <label
        className="af-form__label"
        htmlFor="cjvhuyr8v0000jctd801b4zke"
      >
        <span
          className="af-form__description"
        >
          Internal
        </span>
      </label>
    </div>
    <div
      className="af-form__radio-custom"
    >
      <input
        autoFocus={null}
        checked={false}
        className="af-form__input-radio"
        disabled={false}
        id="cjvhuyr8v0001jctd7c0k7cti"
        name="choice"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        tabIndex={null}
        type="radio"
        value="8762b6dc-9fe3-4c5d-b810-2030e6b36fb5"
      />
      <label
        className="af-form__label"
        htmlFor="cjvhuyr8v0001jctd7c0k7cti"
      >
        <span
          className="af-form__description"
        >
          Virtual
        </span>
      </label>
    </div>
    <div
      className="af-form__radio-custom"
    >
      <input
        autoFocus={null}
        checked={false}
        className="af-form__input-radio"
        disabled={false}
        id="cjvhuyr8v0002jctd4m6a7n3c"
        name="choice"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        tabIndex={null}
        type="radio"
        value="b4d2d5e5-7ece-4dbd-b375-67389b7c4f84"
      />
      <label
        className="af-form__label"
        htmlFor="cjvhuyr8v0002jctd4m6a7n3c"
      >
        <span
          className="af-form__description"
        >
          Override
        </span>
      </label>
    </div>
    <div
      className="af-form__radio-custom"
    >
      <input
        autoFocus={null}
        checked={false}
        className="af-form__input-radio"
        disabled={false}
        id="cjvhuyr8v0003jctdhc3rdo71"
        name="choice"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        tabIndex={null}
        type="radio"
        value="72010226-6d1f-4354-b57d-15953292dbd6"
      />
      <label
        className="af-form__label"
        htmlFor="cjvhuyr8v0003jctdhc3rdo71"
      >
        <span
          className="af-form__description"
        >
          Sealed
        </span>
      </label>
    </div>
    <div
      className="af-form__radio-custom"
    >
      <input
        autoFocus={null}
        checked={false}
        className="af-form__input-radio"
        disabled={false}
        id="cjvhuyr8v0004jctdh7zuf55u"
        name="choice"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        tabIndex={null}
        type="radio"
        value="72010226-6d1f-4354-b57d-15953292dbd6"
      />
      <label
        className="af-form__label"
        htmlFor="cjvhuyr8v0004jctdh7zuf55u"
      >
        <span
          className="af-form__description"
        >
          I don't know
        </span>
      </label>
    </div>
  </div>
</div>
`;
